lib/src/components/NxDrawer/NxDrawer.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, {
+  HTMLAttributes,
+  ReactNode,
+  useContext,
+  useEffect,
+  useRef,
+  useState
+} from 'react';
+import classnames from 'classnames';
+
+import AbstractDialog from '../AbstractDialog/AbstractDialog';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import withClass from '../../util/withClass';
+
+import { Props, propTypes } from './types';
+
+import './NxDrawer.scss';
+
+interface NxDrawerHeaderProps extends HTMLAttributes<HTMLElement>{
+  children: ReactNode;
+}
+
+interface NxDrawerContextValue {
+  closeDrawer: () => void;
```

Comment:
Types should be in `types.ts`

Reply:
> Types should be in `types.ts`

Moved it here: f753e8653362f7f0f0b08d1288dbd2b8d94e5de3